### PR TITLE
[Merged by Bors] - feat: When `s ×ˢ t` is finite

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/18674

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Finite.lean [hunk 1]
```diff
@@ -874,26 +889,64 @@ theorem finite_le_nat (n : ℕ) : Set.Finite { i | i ≤ n } :=
...
+  fun h => hs <| h.of_prod_left ht
+#align set.infinite.prod_left Set.Infinite.prod_left
+
+protected theorem Infinite.prod_right (ht : t.Infinite) (hs : s.Nonempty) : (s ×ˢ t).Infinite :=
+  fun h => ht <| h.of_prod_right hs
+#align set.infinite.prod_right Set.Infinite.prod_right
+
+protected theorem infinite_prod :
+    (s ×ˢ t).Infinite ↔ s.Infinite ∧ t.Nonempty ∨ t.Infinite ∧ s.Nonempty :=
+  by
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Finite.lean:921]: nit: line-wrap the `by`

File: Mathlib/Data/Set/Finite.lean [hunk 2]
```diff
@@ -874,26 +889,64 @@ theorem finite_le_nat (n : ℕ) : Set.Finite { i | i ≤ n } :=
...
+#align set.finite_prod Set.finite_prod
+
+protected theorem Finite.offDiag {s : Set α} (hs : s.Finite) : s.offDiag.Finite := by
   classical
     cases hs
     apply Set.toFinite
 #align set.finite.off_diag Set.Finite.offDiag
 
-theorem Finite.image2 (f : α → β → γ) {s : Set α} {t : Set β} (hs : s.Finite) (ht : t.Finite) :
+protected  theorem Finite.image2 (f : α → β → γ) (hs : s.Finite) (ht : t.Finite) :
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Finite.lean:941]:
```suggestion
protected theorem Finite.image2 (f : α → β → γ) (hs : s.Finite) (ht : t.Finite) :
```

File: Mathlib/Data/Set/Pointwise/Finite.lean [hunk 3]
```diff
@@ -121,6 +127,45 @@ theorem Finite.vsub (hs : s.Finite) (ht : t.Finite) : Set.Finite (s -ᵥ t) :=
...
+#align set.finite_smul_set Set.finite_smul_set
+#align set.finite_vadd_set Set.finite_vadd_set
+
+@[simp, to_additive]
+theorem infinite_smul_set : (a • s).Infinite ↔ s.Infinite :=
+  infinite_image_iff <| (MulAction.injective _).InjOn _
+#align set.infinite_smul_set Set.infinite_smul_set
+#align set.infinite_vadd_set Set.infinite_vadd_set
+
+alias finite_smul_set ↔ finite.of_smul_set _
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Pointwise/Finite.lean:159]: Shouldn't this be:
```suggestion
alias finite_smul_set ↔ Finite.of_smul_set _
```
and similarly below

**Review by eric-wieser — APPROVED** [review]
bors d+

I think there are some case errors in `alias`es.

**Comment by YaelDillies** [comment]
bors merge

**Comment by eric-wieser** [comment]
bors r-

bors d+

There's a CI failure

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**